src/compiler/expression/array.rs

```diff
@@ -12,12 +12,26 @@ use crate::{
 
 #[derive(Debug, Clone, PartialEq)]
 pub struct Array {
+    resolved: Option<Value>,
```

Comment:
How hard would it be to separate out the changes to cache the resolved value of these array and object expressions? On first reading, I thought this was part of the Value CoW optimization and then realized it was doing something different.

Reply:
Sure. I just wanted to show bigger gains ;)